### PR TITLE
[WIP] DeepFool for batches of images

### PR DESCRIPTION
**Pull request by iArunava**

### CODE REVIEW

File: scratchai/attacks/attacks/deepfool.py [hunk 1]
```diff
@@ -38,47 +38,63 @@ def deepfool(x, net:nn.Module, eta:float=0.02, tnc:int=10, miter:int=50,
...
 
     for c in range(1, tnc):
       zgrad(x_idt)
-      logits[psort[c]].backward(retain_graph=True)
+      for ii in range(bs): logits[ii, psort[ii, c]].backward(retain_graph=True)
       cgrad = x_idt.grad.data.cpu().numpy().copy()
 
       # Get new wk and fk
       wk = cgrad - ograd
-      fk = (logits[c] - logits[tlabl]).item()
```
Comment (iArunava) [inline scratchai/attacks/attacks/deepfool.py]: This is a silent bug. The correct implementation should be
`fk = (logits[psort[c]] - logits[tlabl]).item()`
as we need the index of the most closest `c`th class. And instead `logits[c]` will simply take the `c`th index of the logit (or the `c`th class) irrespective of which class is the closest based on the current input. 
**This is fixed in this PR** with
`fk = (logits[lbs, psort[:, c]] - logits[lbs, tlabl]) \`
as can be seen, its updated to take the index from `psort[:, c]`
